### DISCUSSION TITLE
Remove truck stops.

### DISCUSSION THREAD

**Discussion by singetail246** [voxeltycoon/issues#748] [General]
Now hear me out: I have seen many people complain how stores cannot grab items from warehouses in their radius and how they think it's stupid that a truck has to deliver items to a truck stop in order to sell them to the store, even if it's grabbing them from the same truck stop.

My suggestion is to remove the normal truck stops and instead give warehouses and stores attached truck stops.  So when you place a warehouse, there's a truck stop there. And when a store spawns, it creates a truck stop with it. I think it would make it a lot easier to see you need to deliver the items to the store's truck stop in order to sell the items.

Also, in a realistic point of view it makes sense. Trucks go to warehouses in real life to pick up the cargo, and drop them off at the store. They don't go to some stop two streets away.

**Comment by victorleblais** [comment]
What's the difference between that and building truck stops manually? (truck stop near shop + truck stop near warehouse)

Also, you do want to have truck stops without attached warehouses in some situation (selling to a shop is the biggest example, but also making a dedicated factory area). And when I build truck stops myself, I can decide where to put them, whereas an automatic truck stop attached to a shop would not give me similar freedom.

And you will still have people that build a warehouse near a shop and complain that the items in the warehouse don't go into the shop if they're in range with each other.

I'm not sure how it would be an improvement: it just removes gameplay.

**Reply by singetail246** [reply]
To me, the main confusion seems to be that people think the truck sends items to the stop, which in turn sends them to the shop, not realizing it is the truck that is actually sending items to the shop.  And so they think if the shop can grab items from the stop, why can't they grab from warehouses also.  

If there is a truck stop attached to the shop, it will help visualize to players that "hey, items need to be delivered HERE to the shop in order for the shop to receive them"

As to the removing of the truck stop, I didn't see a reason to have a truck stop if shops and warehouses have them. But if there are ways people use them other than those reasons, then by all means keep them. My main idea here is to give shops stops to help the players realize that the shop is not proactively grabbing items from buildings in range, but instead waiting for items from the trucks. This area of the game is annoying quite a bit of players on the Steam Forums and I am offering my idea of a solution. Stops attached to shops may affect the flexibility slightly, but I think it would be worthwhile if it helps clear up the confusion many people have.

**Comment by DBT85** [comment]
I use refreshfr's mod with small stops to basically do this. My feeling is that nothing should automatically teleport from a station a block away to a store.
